python/sglang/srt/managers/scheduler.py

```diff
@@ -1017,7 +1017,8 @@ def process_batch_result(self, batch: ScheduleBatch, result):
             self.process_batch_result_prefill(batch, result)
         elif batch.forward_mode.is_dummy_first():
             batch.next_batch_sampling_info.update_regex_vocab_mask()
-            self.current_stream.synchronize()
+            if self.device != "cpu":
+                self.current_stream.synchronize()
```

Comment:
Can you skip this check by making `self.current_stream.synchronize` a dummy function?
Basically we would like to avoid too many `if/else` in the core logic, but we can put them during initialization.

Reply:
Updated the code in https://github.com/sgl-project/sglang/pull/2806/commits/a6473e6aa1ed32a2ef7ece5f7578394c93667d90. Could you please take another look if this is okay?